src/paraswap.ts

```diff
@@ -426,13 +427,18 @@ export class ParaSwap {
     userAddress: Address,
     tokenAddress: Address,
     _provider?: any,
+    contractOptions?: ContractOptions,
   ): Promise<string> {
     return new Promise(async (resolve, reject) => {
       const spender = await this.getSpender();
 
       const provider = _provider || this.web3Provider;
 
-      const contract: any = new provider!.eth.Contract(ERC20_ABI, tokenAddress);
+      const contract: any = new provider!.eth.Contract(
+        ERC20_ABI,
+        tokenAddress,
+        contractOptions,
+      );
 
       return contract.methods
         .approve(spender, amount)
```

Comment:
I think it's better to pass in `SendOptions` for `contract.methods.approve.send({...options, from})`, then ContractOptions for the whole contract
Types are more specific to transaction + [ContractOptions includes `data`](https://web3js.readthedocs.io/en/v1.3.4/web3-eth-contract.html#parameters) which we don't need

Reply:
Agree, and fixed ;)